newIDE/app/src/GameDashboard/Monetization/UserEarningsWidget.js

```diff
@@ -0,0 +1,252 @@
+// @flow
+import { Trans } from '@lingui/macro';
+import * as React from 'react';
+
+import { LineStackLayout, ResponsiveLineStackLayout } from '../../UI/Layout';
+import Text from '../../UI/Text';
+import { Spacer } from '../../UI/Grid';
+import BackgroundText from '../../UI/BackgroundText';
+import Link from '../../UI/Link';
+import Window from '../../Utils/Window';
+import RaisedButton from '../../UI/RaisedButton';
+import Coin from '../../Credits/Icons/Coin';
+import AuthenticatedUserContext from '../../Profile/AuthenticatedUserContext';
+import PlaceholderError from '../../UI/PlaceholderError';
+import { Tooltip } from '@material-ui/core';
+import CreditOutDialog from './CashOutDialog';
+import GDevelopThemeContext from '../../UI/Theme/GDevelopThemeContext';
+import DashboardWidget from '../Widgets/DashboardWidget';
+import { useResponsiveWindowSize } from '../../UI/Responsive/ResponsiveWindowMeasurer';
+
+const styles = {
+  separator: {
+    height: 50,
+  },
+  buttonContainer: {
+    display: 'flex',
+    flexDirection: 'column',
+    alignItems: 'stretch',
+  },
+};
+
+type Props = {|
+  fullWidth?: boolean,
+|};
+
+const UserEarningsWidget = ({ fullWidth }: Props) => {
+  const {
+    userEarningsBalance,
+    onRefreshEarningsBalance,
+    onRefreshLimits,
+  } = React.useContext(AuthenticatedUserContext);
+  const theme = React.useContext(GDevelopThemeContext);
+  const { isMobile } = useResponsiveWindowSize();
+
+  const [earningsInMilliUsd, setEarningsInMilliUsd] = React.useState(0);
+  const [earningsInCredits, setEarningsInCredits] = React.useState(0);
+  const [error, setError] = React.useState(null);
+  const intervalValuesUpdate = React.useRef(null);
+
+  const [selectedCashOutType, setSelectedCashOutType] = React.useState<
+    ?'cash' | 'credits'
+  >(null);
+
+  const fetchUserEarningsBalance = React.useCallback(
```

Comment:
This looks like ill-named. Should it be `animateEarnings` ?

Reply:
indeed, probably an old naming. I'll fix that